packages/core/src/api.ts

```diff
@@ -239,56 +199,15 @@ export type CompatibleDeployFunc = (
   checkOnly?: boolean,
 ) => Promise<DeployResult>
 
-const getDeployArgs: (
-  workspaceOrParams: Workspace | DeployParams,
-  actionPlan?: Plan,
-  reportProgress?: ReportProgress,
-  accounts?: string[],
-  checkOnly?: boolean,
-) => DeployParams = (workspaceOrParams, actionPlan, reportProgress, accounts, checkOnly) => {
-  if ('adapterCreators' in workspaceOrParams) {
-    return workspaceOrParams
-  }
-  if (actionPlan === undefined || reportProgress === undefined) {
-    throw new Error('actionPlan and reportProgress should not be undefined if workspace is a workspace')
-  }
-  return {
-    workspace: workspaceOrParams,
-    accounts,
-    checkOnly,
-    actionPlan,
-    reportProgress,
-    adapterCreators: allAdapterCreators,
-  }
-}
-
-// As a transitionary step, we support both a workspace input and an argument object
-export function deploy(args: DeployParams): Promise<DeployResult>
-// @deprecated
-export function deploy(
-  inputWorkspace: Workspace,
-  inputActionPlan?: Plan,
-  inputReportProgress?: ReportProgress,
-  inputAccounts?: string[],
-  inputCheckOnly?: boolean,
-): Promise<DeployResult>
-
-export async function deploy(
-  inputWorkspace: Workspace | DeployParams,
-  inputActionPlan?: Plan,
-  inputReportProgress?: ReportProgress,
-  inputAccounts?: string[],
-  inputCheckOnly = false,
-): Promise<DeployResult> {
-  // for backward compatibility
+export async function deploy(args: DeployParams): Promise<DeployResult> {
```

Comment:
nit - more conventional to unpack the arguments in the function
```suggestion
export async function deploy({workspace, actionPlan, reportProgress, accounts = workspace.accounts(), checkOnly = false, adapterCreators}: DeployParams): Promise<DeployResult> {
```

same for other functions